pkg/cmd/experimental/project/new_project.go

```diff
@@ -0,0 +1,117 @@
+package project
+
+import (
+	"fmt"
+
+	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	"github.com/spf13/cobra"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/client"
+	projectapi "github.com/openshift/origin/pkg/project/api"
+)
+
+type newProjectOptions struct {
+	projectName  string
+	displayName  string
+	description  string
+	clientConfig clientcmd.ClientConfig
+
+	adminRole             string
+	masterPolicyNamespace string
+	adminUser             string
+}
+
+func NewCmdNewProject(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	options := &newProjectOptions{clientConfig: clientConfig}
+
+	cmd := &cobra.Command{
+		Use:   "new-project <project-name>",
+		Short: "create a new project",
+		Long:  `create a new project`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if !options.complete(cmd) {
+				return
+			}
+
+			err := options.run()
+			if err != nil {
+				fmt.Printf("%v\n", err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&options.masterPolicyNamespace, "master-policy-namespace", "master", "master policy namespace")
+	cmd.Flags().StringVar(&options.adminRole, "admin-role", "admin", "project admin role name in the master policy namespace")
+	cmd.Flags().StringVar(&options.adminUser, "admin", "", "project admin username")
+	cmd.Flags().StringVar(&options.displayName, "display-name", "", "project display name.  defaults to project-name")
+	cmd.Flags().StringVar(&options.description, "description", "", "project description.  defaults to display-name")
+
+	return cmd
+}
+
+func (o *newProjectOptions) complete(cmd *cobra.Command) bool {
+	args := cmd.Flags().Args()
+	if len(args) != 1 {
+		cmd.Help()
+		return false
+	}
+
+	o.projectName = args[0]
+
+	if len(o.displayName) == 0 {
```

Comment:
on further reflection, I don't think we should default the displayName to projectName either (at least not client-side). That defaulting should happen server-side if we want it. The UI falls back to the projectName if displayName is empty.